docs/download.md

```diff
@@ -49,10 +49,16 @@ The plugin is tested on the following architectures:
 		Databricks 10.4 ML LTS (GPU, Scala 2.12, Spark 3.2.1)
 		Databricks 11.3 ML LTS (GPU, Scala 2.12, Spark 3.3.0)
 		Databricks 12.2 ML LTS (GPU, Scala 2.12, Spark 3.3.2)
+		Databricks 13.3 ML LTS (GPU, Scala 2.12, Spark 3.4.1)
```

Comment:
We removed Databricks 13.3 support for v23.12.0 release. Refer to https://github.com/NVIDIA/spark-rapids/pull/9890

Reply:
updated, thank you!